project2/src/components/FormGroup/index.jsx

```diff
@@ -0,0 +1,30 @@
+import { Form } from "react-bootstrap";
+
+function FormGroup({
+  label,
+  id,
+  type,
+  name,
+  placeholder,
+  value,
+  onChange,
+  error,
+}) {
```

Comment:
Em có thể truyền như này
```suggestion
function FormGroup({
  label,
  id,
  type,
  onChange,
  ...rest
}) {
```
Những thứ gì bắt buộc truyền thì em có thể giữ nguyên. Còn những attr có thể có hoặc không như placeholder, error,... thì em có thể để dưới dạng spread operator. Ở đây là `...rest`
Lúc này 
```
<Form.Control
        id={id}
        type={type}
        onChange={onChange}
        {...rest}
/>
```

Khi sử dụng thì có thể truyền như bình thường, nhưng nó có thể truyền thêm những attr khác
```
 <FormGroup
    label="Họ tên*"
    data-label="Họ tên*"
    id="username"
    type="text"
    placeholder="Họ tên"
    value={formik.values.username}
    onChange={formik.handleChange}
    error={formik.errors.username}
/>
```